AEPMessaging/Sources/Messaging.swift

```diff
@@ -454,21 +506,24 @@ public class Messaging: NSObject, Extension {
         let requestedSurfaces = surfaces.filter { $0.isValid }
 
         guard !requestedSurfaces.isEmpty else {
-            Log.debug(label: MessagingConstants.LOG_TAG, "Unable to retrieve feed messages, no valid surface paths found.")
+            Log.debug(label: MessagingConstants.LOG_TAG, "Unable to retrieve propositions, no valid surface paths found.")
             dispatch(event: event.createErrorResponseEvent(AEPError.invalidRequest))
             return
         }
 
-        // get propositions from rules engine where conditions are met by this event
-        let ruleConsequencePropositions = getPropositionsFromFeedRulesEngine(event)
-        // get cached propositions
-        var requestedPropositions = retrieveCachedPropositions(for: requestedSurfaces)
-
-        for (surface, propositions) in ruleConsequencePropositions {
-            requestedPropositions.addArray(propositions, forKey: surface)
+        // get requested content cards from cache
+        let requestedContentCards = contentCardsBySurface.filter { surfaces.contains($0.key) }
+        
+        // get requested propositions (cbe) from cache
+        let requestedPropositions = retrieveCachedPropositions(for: requestedSurfaces)
+                
+        // merge the two maps
+        var mergedPropositions = requestedContentCards
+        for (surface, propositions) in requestedPropositions {
+            mergedPropositions.addArray(propositions, forKey: surface)
```

Comment:
This is called from inside `OperationOrderer` backed by a DispatchQueue, is it possible to be called from multiple threads at the same time? If yes, we should make it a thread safe map

Reply:
operation orderer only processes one at a time